### DISCUSSION TITLE
Quasar v1->v2 upgrade and reactivity issue

### DISCUSSION THREAD

**Discussion by MarakJaroslav** [quasarframework/quasar#11726] [Show and Tell]
Hi there!
I am trying to upgrade my vuex files from Quasar v1 to latest v2 (vuejs 3) and i have still some problem with reactivity.
In origin v1 code I have this two vuex mutations which instantly caused DOM update and rerendering GUI in app:

    putImage(state, payload) {
        Vue.set(state.images, payload.image, payload.url)
    },

    removeImage(state, fileName) {
        Vue.delete(state.images, fileName)
    },

And because .set() and .delete() methods has been removed from Global API in Vue 3, I tried to apply reactivity on state object:

    import { reactive } from 'vue'

    const state = reactive({
       images : {}
    })

    putImage(state, payload) {
        state.images[payload.image] = payload.url
    },

    removeImage(state, fileName) {
        delete state.images[fileName]
    },

But this approach is not working. After any update on images object vue is not rerendered automatically. 
So my question is, how to do automatic rerendering after any update in vuex like in Quasar V1?

Thanky You!
J.

**Comment by MilosPaunovic** [comment]
Have you followed guides for migration:
[https://quasar.dev/start/upgrade-guide#vuex-v4](https://quasar.dev/start/upgrade-guide#vuex-v4)
[https://next.vuex.vuejs.org/guide/migrating-to-4-0-from-3-x.html](https://next.vuex.vuejs.org/guide/migrating-to-4-0-from-3-x.html)

Is there a particular reason why aren't you storing your images in an object with simple assignment of a value?

**Comment by hawkeye64** [comment]
@MarakJaroslav Use the `Array.prototype.splice` function to manipulate arrays and maintain reactivity.
